pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -919,3 +815,29 @@ func MakeNextPodFunc(queue SchedulingQueue) func() *v1.Pod {
 func podInfoKeyFunc(obj interface{}) (string, error) {
 	return cache.MetaNamespaceKeyFunc(obj.(*podInfo).pod)
 }
+
+// IsIdenticalErrorPodExists returns false when the error pod cannot be found in the
+// in the unschedulableQ or podBackoffQ. Otherwise, this function continues to compare
+// the found pod with the input pod, and returns error message if the two pods have 
+// any difference. This function currently is only for test purpose. 
+func IsIdenticalErrorPodExists(p *PriorityQueue, pod *v1.Pod) (bool, error) {
```

Comment:
Replace this with a simplified "Get" or "Has" function. For example:

```go
// Has returns true if a pod is held by any scheduler queue
// and it returns err if an internal error happens
func (p *PriorityQueue) Has(pod *v1.Pod) (bool, error) {
	p.lock.RLock()
	defer p.lock.RUnlock()
	pInfo := p.newPodInfo(pod)
	if _, exists, err := p.activeQ.Get(pInfo); err == nil {
		if exists {
			return exists, nil
		}
	} else {
		return false, err
	}
	if _, exists, err := p.podBackoffQ.Get(pInfo); err == nil {
		if exists {
			return exists, nil
		}
	} else {
		return false, err
	}
	return p.unschedulableQ.get(pod) != nil, nil
}
```

Reply:
In addition, the Error pod in the test will not have chance to be in activeQ. There is a little bit  unmatched if we test the error func with a function finding pod in all of three queues.

Reply:
In unit test, you're right. But I think it's trivial. I'm fine with either, you can remove checking for activeQ, and rename the method name.

Reply:
ok. I modified it to only get pod from unschedulableQ as there is no chance to go backoffQ due to no moveRequest happen in the factory's unit test function.